### DISCUSSION TITLE
Export Physics ?

### DISCUSSION THREAD

**Discussion by klauskobald** [atteneder/glTFast#272] [Q&A]
I made a simple test with 2 Joints and 3 rigid bodies.
But when exporting and opening in https://gltf-viewer.donmccurdy.com/ I only get a static model.
Would physics even work?

**Comment by atteneder** [comment]
Great topic! Thanks for raising awareness!

The core glTF specification 2.0 does not contain physics properties and as far as I know there is no ratified extension about it either.

However, I'm sure there are draft extensions already and it's only a matter of time until some basic physics extensions will be ratified.

I can even imagine glTFast taking a leap and becoming one of the first implementors of such an experimental extension, but not in the very near future.

As always, I'm open to discussions and contributions.

**Reply by klauskobald** [reply]
But animiations would work, right?
I have seen moving 3d objects on opensea.io.

**Reply by atteneder** [reply]
Yes, animating node transforms and morph targets (blend shapes) will get added! Currently only import is supported when it comes to animation.

**Reply by klauskobald** [reply]
bummer - I only need export! Any plans for that?

**Reply by atteneder** [reply]
Gut feeling is early 2022. End of 2021 in the best case.

**Reply by mikeskydev** [reply]
Some (in draft) extension work for rigid body support happening here: https://github.com/omigroup/gltf-extensions/pull/75 Feedback always helpful!

**Comment by MattGen** [comment]
> Gut feeling is early 2022. End of 2021 in the best case.

Hello, have you been working on that ? I'm using your exporter and it seem to be the easiest one out there and I'm thankful for that, but animations is what I'm looking to export.
